src/ThriveWorkerUnit.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./IBadgeQuery.sol";
+
+/**
+ * @title ThriveWorkerUnit
+ * @dev Contract of a work unit representing a task where contributors complete and earn rewards.
+ */
+contract ThriveWorkerUnit {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    address public immutable moderator;
+    address public immutable rewardToken;
+    uint256 public immutable rewardAmount;
+    uint256 public immutable maxRewards;
+    uint256 public immutable validationRewardAmount;
+    uint256 public maxCompletionsPerUser;
+    uint256 public deadline;
+
+    IBadgeQuery public badgeQuery;
+    bool public ready;
+
+    string public validationMetadata;
+    string public metadataVersion;
+    string public metadata;
+
+    EnumerableSet.AddressSet private validators;
+    EnumerableSet.Bytes32Set private requiredBadges;
+    mapping(address => uint256) public completions;
+
+    event Initialized();
+    event ConfirmationAdded(
+        address indexed contributor,
+        string validationMetadata,
+        uint256 rewardAmount,
+        address indexed validator
+    );
+
+    event ConfigurationUpdated(string field, uint256 newValue);
+    event MetadataUpdated(string field, string newValue);
```

Comment:
hmm I think we can remove these 2 events, even when implementing setters they're not much of use